packages/code-studio/src/styleguide/colorUtils.ts

```diff
@@ -0,0 +1,142 @@
+import { ColorUtils } from '@deephaven/utils';
+import { useLayoutEffect, useRef } from 'react';
+
+export const INVALID_COLOR_BORDER_STYLE =
+  '2px solid var(--dh-color-notice-default-bg)';
+
+/** Return black or white contrast color */
+export function contrastColor(color: string): 'black' | 'white' {
+  const rgba = ColorUtils.parseRgba(ColorUtils.asRgbOrRgbaString(color) ?? '');
+  if (rgba == null || rgba.a < 0.5) {
+    return 'white';
+  }
+
+  const { r, g, b } = rgba;
+  const y = (299 * r + 587 * g + 114 * b) / 1000;
+  return y >= 128 ? 'black' : 'white';
```

Comment:
Use `ColorUtils.getBrightness` here:
```suggestion
  return ColorUtils.getBrightness([r, g, b]) >= 128 ? 'black' : 'white';
```

Reply:
Addressed in [ColorUtils.getBrightness](https://github.com/deephaven/web-client-ui/pull/1651)